File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/discount/BogoDiscount.java
package com.fixture.shop.discount;

import com.fixture.shop.models.Order;
import com.fixture.shop.models.OrderItem;

/**
 * Buy-one-get-one — cheapest item is free.
 */
public class BogoDiscount implements DiscountStrategy {

    @Override
    public double apply(Order order) {
        if (order.getItems().size() < 2) {
            return order.rawTotal();
        }
        double cheapest = Double.MAX_VALUE;
        for (OrderItem item : order.getItems()) {
            if (item.subtotal() < cheapest) {
                cheapest = item.subtotal();
            }
        }
        return order.rawTotal() - cheapest;
    }
}

File: skill_sdk/tests/fixtures/ts-project/src/shop/discount/BogoDiscount.ts
import { DiscountStrategy } from './DiscountStrategy';
import { Order } from '../models/Order';
import { OrderItem } from '../models/OrderItem';

/**
 * Buy-one-get-one — cheapest item is free.
 */
export class BogoDiscount implements DiscountStrategy {
  apply(order: Order): number {
    if (order.getItems().length < 2) {
      return order.rawTotal();
    }
    let cheapest = Number.MAX_VALUE;
    for (const item of order.getItems()) {
      if (item.subtotal() < cheapest) {
        cheapest = item.subtotal();
      }
    }
    return order.rawTotal() - cheapest;
  }
}
